libipl/p10/ipl0.C

```diff
@@ -374,28 +387,137 @@ static int ipl_alignment_check(void)
 	return -1;
 }
 
+static int initialize_and_check_clock_chip(uint8_t& clock_count)
+{
+	struct pdbg_target *clock_target;
+	enum pdbg_target_status status;
+	uint8_t data;
+	int i2c_rc = 0;
+	int rc = 0;
+
+	// initialize output variable.
+	clock_count = 0;
+
+	ipl_log(IPL_DEBUG, "Istep: soft reset clock, and verify clock status register\n");
+	pdbg_for_each_class_target("oscrefclk", clock_target) {
+		if(!ipl_is_functional(clock_target)){
+			continue;
+		}
+
+		// clock count has to be incremented even if it failed to initiallize
+		clock_count++;
+
+		status = pdbg_target_probe(clock_target);
+		if(status != PDBG_TARGET_ENABLED){
+			ipl_log(IPL_ERROR, "pdbg_target_probe is failed for clock '%s', with status = %d\n",
+				pdbg_target_path(clock_target), status);
+
+			ipl_error_callback(IPL_ERR_TGT_PROBE_FAILED);
+			rc++;
+			continue;
+		}
+
+		// Resetting the clock chip, so that it will recalibrate input oscillator
+		// signal and identifies bad oscillators.
+		data = OSC_RESET_CMD;
+		i2c_rc = i2c_write(clock_target, 0, OSC_CTL_OFFSET, OSC_RESET_CMD_LEN, &data);
+		if(i2c_rc){
+			ipl_log(IPL_ERROR, "soft reset command is failed for clock '%s' with rc = %d\n",
+				pdbg_target_path(clock_target), rc);
+
+			ipl_error_callback(IPL_ERR_CLK_FAILED);
+			rc++;
+			continue;
+		}
+		else{
+			ipl_log(IPL_DEBUG, "soft reset command is successfull for clock '%s'\n",
+				pdbg_target_path(clock_target));
+		}
+
+		// wait for clock to restart
+		std::this_thread::sleep_for(std::chrono::milliseconds(CLOCK_RESTART_DELAY_IN_MS));
```

Comment:
This is max delay or minimum delay required to complete reset?
Considering ipl performace this is not good design direction waiing for max value instead  check smaller intervals and continnue.

Reply:
This is the maximum expected time to clock to restart and hence minimum delay required.
we are waitng only 100 ms.  do you think its a problem?

If i want to poll this, what is your suggestion on polling interval?

Reply:
if it is total 100ms and max for 2 cllock 200ms , i think we are good here.

Reply:
yes, its maximu 200 ms.